### DISCUSSION TITLE
Is the nomad scheduler plugin still viable?

### DISCUSSION THREAD

**Discussion by bitnom** [dokku/dokku#6694] [Q&A]
I see the nomad scheduler plugin hasn't received a commit in 5 years. I'm trying to decide between k3s and nomad, and the ability to scale dokku is one of my considerations. Am I basically going to have a much better experience with k3s if I plan to cluster dokku?

https://github.com/dokku/dokku-scheduler-nomad

**Comment by josegonzalez** [comment]
The current built-in k3s functionality was [sponsored by Rechat](https://rechat.com/). The existing kubernetes functionality didn't quite fit their needs, and so  I consulted with them on requirements and that more or less guided where the feature-set is today. Since it's in the core, its more likely that I'll fix things/implement functionality, but outside of obvious defects, a lot of that will be guided by the community/sponsors. Regardless, I think it's in a pretty good spot for folks trying to dip their toes into clustered Dokku installations.

I ran a few Nomad clusters in production and Nomad is near and dear to my heart. I'd love to see Nomad get the same push internally, but to be quite honest, we probably wouldn't have k3s integration had it not been sponsored work. Integration with a third-party scheduling tool is not quite straightforward, and the Nomad integration definitely needs a bit of polish for me to consider moving it into the core.

That said, I think the Nomad integration should mostly work and I'd be more than happy to fix any obvious bugs. If you have pull requests implementing functionality, those would be welcome as well, and I'd be more than happy to help folks contribute to make it core-worthy.

Hope that gives some insight.
